mailet/standard/src/main/java/org/apache/james/transport/mailets/LogMessage.java

```diff
@@ -52,11 +56,17 @@
      * or mark it as finished.
      */
     private final Logger logger;
+    private LogLevel logLevel;
     private boolean passThrough = true;
-    private boolean headers = true;
-    private boolean body = true;
-    private int bodyMax = 0;
-    private String comment = null;
+    private boolean headers;
+    private boolean body;
+    private int bodyMax;
+    private String comment;
+    private List<String> specificHeaderNames;
+    private List<String> specificAttributeNames;
+
+    //    private StringBuffer logBuffer = new StringBuffer();
+    private ThreadLocal<StringBuilder> logBuffer = new ThreadLocal<>();
```

Comment:
Or.... I am really not confortable with statefull mailet.

Maybe a better way of seeing things is that each `log*` method return a String (representing logParts) that gets then concatenated together & logged.

Mutability makes reasoning about programming hard. We tend to prefer predictable pure function...